pennylane_lightning/lightning_qubit.py

```diff
@@ -813,32 +896,46 @@ def expval(self, observable, shot_range=None, bin_size=None):
             samples = self.sample(observable, shot_range=shot_range, bin_size=bin_size)
             return np.squeeze(np.mean(samples, axis=0))
 
-        # Initialization of state
-        ket = np.ravel(self._pre_rotated_state)
+        batched_expval = []
 
-        state_vector = StateVectorC64(ket) if self.use_csingle else StateVectorC128(ket)
-        M = MeasuresC64(state_vector) if self.use_csingle else MeasuresC128(state_vector)
-        if observable.name == "SparseHamiltonian":
-            if Kokkos_info()["USE_KOKKOS"] == True:
-                # converting COO to CSR sparse representation.
-                CSR_SparseHamiltonian = observable.data[0].tocsr(copy=False)
-                return M.expval(
-                    CSR_SparseHamiltonian.indptr,
-                    CSR_SparseHamiltonian.indices,
-                    CSR_SparseHamiltonian.data,
+        batch_size, batched_states = self.check_and_guarantee_batched_states_shape(
+            self._pre_rotated_state
+        )
+        for state in batched_states:
+            state = np.ravel(state)
+
+            state_vector = StateVectorC64(state) if self.use_csingle else StateVectorC128(state)
+            M = MeasuresC64(state_vector) if self.use_csingle else MeasuresC128(state_vector)
+            if observable.name == "SparseHamiltonian":
+                if Kokkos_info()["USE_KOKKOS"] == True:
+                    # converting COO to CSR sparse representation.
+                    CSR_SparseHamiltonian = observable.data[0].tocsr(copy=False)
+                    batched_expval += [
+                        M.expval(
+                            CSR_SparseHamiltonian.indptr,
+                            CSR_SparseHamiltonian.indices,
+                            CSR_SparseHamiltonian.data,
+                        )
+                    ]
+                    continue
+                raise NotImplementedError(
+                    "The expval of a SparseHamiltonian requires Kokkos and Kokkos Kernels."
                 )
-            raise NotImplementedError(
-                "The expval of a SparseHamiltonian requires Kokkos and Kokkos Kernels."
-            )
 
-        if observable.name in ["Hamiltonian", "Hermitian"]:
-            ob_serialized = _serialize_ob(observable, self.wire_map, use_csingle=self.use_csingle)
-            return M.expval(ob_serialized)
+            if observable.name in ["Hamiltonian", "Hermitian"]:
+                ob_serialized = _serialize_ob(
+                    observable, self.wire_map, use_csingle=self.use_csingle
+                )
+                batched_expval.append(M.expval(ob_serialized))
+                continue
 
-        # translate to wire labels used by device
-        observable_wires = self.map_wires(observable.wires)
+            # translate to wire labels used by device
+            observable_wires = self.map_wires(observable.wires)
```

Comment:
Does this mean all the states should be observed with the same wires now? If yes, do we need to explictly mention it in docstring?

Reply:
Sorry, I'm not exactly getting what you are asking. We, in principle, have the same behaviour for broadcasted parameters + measurements as before. 
Does that answer your question?

Reply:
Thanks! Aha, What I mentioned above should be a parameter "scatter" behaviour.